portfolio/src/main/webapp/script.js

```diff
@@ -81,29 +82,78 @@ function createCommentElement(comment) {
   const commentInputElement = document.createElement('p');
   commentInputElement.innerText = comment.commentInput;
 
+  const deleteButtonElement = document.createElement('button');
+  deleteButtonElement.innerText = 'Delete';
+  deleteButtonElement.addEventListener('click', () => {
+    deleteComment(comment);
+    commentBlock.remove();
+  });
+
   commentBlock.appendChild(nameElement);
   commentBlock.appendChild(commentInputElement);
+  commentBlock.appendChild(deleteButtonElement);
   commentBlock.appendChild(document.createElement('hr'));
   return commentBlock;
 }
 
-/*
+/** Tells the server to delete the comment. */
+deleteComment = async (comment) => {
+  const params = new URLSearchParams();
+  params.append('id', comment.id);
+  let response = await fetch('/delete-data', {method: 'POST', body: params});
+  getBlogComments(window.localStorage.getItem("comment-limit"));
+}
+
+/** Fetch login status and display comments form or login link accordingly. */
+getLoginStatus = () => {
+  fetch("/login").then(response => response.text()).then((result) => {
```

Comment:
Hm so I did await / async for this function because I didn't really know what to do with the response variable in a fetch.then situation (it's like a dummy b/c we don't actually use it)... but I'm sure we can align on a common implementation for the capstone

Reply:
You can do the await/async for the other ones too even if you need to use the response variable fyi.

OK, sg on aligning for the future project/codebase.